### ISSUE TITLE
[FEATURE] Render Jinja templates with custom variables

### ISSUE THREAD

**Issue by elenamartina** [databrickslabs/dbx#178]
## Expected Behavior
It would be beneficial to specify custom variables or `.yaml` files containing custom variables to render Jinja cluster configuration templates.  
This would allow, for example, to specify job-specific variables for each deployment without having to export them as global environment variables (which, when doing multiple deployments at once, would be shared across different jobs and cause issues).

## Current Behavior
Currently Jinja templates are rendered using environment variables only, as implemented [here](https://github.com/databrickslabs/dbx/blob/743b579a4ac44531f764c6e522dbe5a81a7dc0e4/dbx/utils/common.py#L189):

```python
j2_env = jinja2.Environment(loader=jinja2.FileSystemLoader(file_path))
return j2_env.get_template(file_name).render(os.environ)
```

## Your Environment
* dbx version used: 0.4.1
* Databricks Runtime version: 9.1LTS

**Comment by renardeinside** [comment]
hi @elenamartina , 

could you please take a look at the new include functionality? will it cover the requirement for custom variables in Jinja tempaltes?

**Comment by renardeinside** [comment]
hi @elenamartina , 

now I see your point. So you would like to have a file in .yml/.json format which provides additional environment variables to the Jinja2 template which is used for job creation?

**Comment by elenamartina** [comment]
Yes exactly 😄 the pr is a bit old now and has some conflicts, so might need a bit of refactoring. But apart from that, the work to implement this has been done and tested (and it's currently in Production at Gousto, my company!)

**Comment by renardeinside** [comment]
now I see the value and meaning. I think name of the argument needs to be changed a little bit, because it's hard to guess the meaning of it. I'll take care of this and adopt your PR.

**Comment by elenamartina** [comment]
Thank you so much - sorry for not explaining this well before!

**Comment by renardeinside** [comment]
hi @elenamartina , 
I've slightly re-worked your code to make it more abstract and include it into the main canvas. 
It will be released in 0.6.6 in the upcoming week. Thanks a lot for raising the issue and contributing to it!
